# HW4 Task 2 Fire Perimeter

Author: Haylee Oyler

**Link to github repo:** https://github.com/haylee360/eds220-hwk4 

## About

**Purpose:** This notebook details the steps to select a specific fire perimter for the 2017 Thomas Fire in Santa Barbara and Ventura County. This fire burned over 280,000 acres and was one of the largest fires in the state at the time. 

**Highlights:** 
- Importing and exploring our geospatial fire perimeter data
- Select geospatial data specific to the 2017 Thomas Fire
- Export the Thomas Fire geospatial data to use in our next notebook

**About the Data:**

**References:** [CalFire Fire Perimeter Data](https://www.fire.ca.gov/what-we-do/fire-resource-assessment-program/fire-perimeters) 


## Fire perimeter data retrieval and selection

### Import data and explore
Import the geodatabase containing all fire perimeters in California. 

In [ ]:
# Load libraries
import os
import pandas as pd
import matplotlib.pyplot as plt
import geopandas as gpd

# Change display settings to see all column names
pd.set_option("display.max.columns", None)

# Import fire perimeter data
perimeters = gpd.read_file(os.path.join('no_push_data',
                                    'calfire_all.gdb')
                                    )


In [2]:
# View the first five rows of the perimeter data frame
perimeters.head()

YEAR_ STATE AGENCY UNIT_ID  FIRE_NAME   INC_NUM  \
0  2023.0    CA    CDF     SKU  WHITWORTH  00004808   
1  2023.0    CA    LRA     BTU     KAISER  00010225   
2  2023.0    CA    CDF     AEU    JACKSON  00017640   
3  2023.0    CA    CDF     AEU     CARBON  00018821   
4  2023.0    CA    CDF     AEU    LIBERTY  00018876   

                                  IRWINID                 ALARM_DATE  \
0  {7985848C-0AC2-4BA4-8F0E-29F778652E61}  2023-06-17T00:00:00+00:00   
1  {43EBCC88-B3AC-48EB-8EF5-417FE0939CCF}  2023-06-02T00:00:00+00:00   
2  {B64E1355-BF1D-441A-95D0-BC1FBB93483B}  2023-07-01T00:00:00+00:00   
3  {CB41DB0A-E4B1-489D-A4EA-738F2CD6DB3B}  2023-07-11T00:00:00+00:00   
4  {F83F70A4-07A7-40B8-BD51-10CCC1C30D63}  2023-07-11T00:00:00+00:00   

                   CONT_DATE  C_METHOD  CAUSE  OBJECTIVE COMPLEX_NAME  \
0  2023-06-17T00:00:00+00:00       1.0    5.0        1.0         None   
1  2023-06-02T00:00:00+00:00       1.0    5.0        1.0         None   
2  2023-07-02T00:00:00+00:00       1.0    2.0        1.0         None   
3  2023-07-11T00:00:00+00:00       1.0    9.0        1.0         None   
4  2023-07-12T00:00:00+00:00       1.0   14.0        1.0         None   

  COMPLEX_ID COMMENTS FIRE_NUM  Shape_Length     Shape_Area  \
0       None     None     None    933.907127   23184.946763   
1       None     None     None    986.343353   55046.898792   
2       None     None     None   2115.572740  112561.147688   
3       None     None     None   1994.866526  237794.210367   
4       None     None     None   4008.305800  287241.810389   

                                            geometry  
0  MULTIPOLYGON (((-243242.555 394267.371, -24324...  
1  MULTIPOLYGON (((-167826.147 203275.612, -16782...  
2  MULTIPOLYGON (((-79176.443 31393.490, -78959.0...  
3  MULTIPOLYGON (((-85936.934 47921.886, -85856.4...  
4  MULTIPOLYGON (((-86427.836 26461.613, -86436.1...

In [3]:
# View information about the perimeter data frame
perimeters.info()

<class 'geopandas.geodataframe.GeoDataFrame'>
RangeIndex: 22261 entries, 0 to 22260
Data columns (total 19 columns):
 #   Column        Non-Null Count  Dtype   
---  ------        --------------  -----   
 0   YEAR_         22184 non-null  float64 
 1   STATE         22261 non-null  object  
 2   AGENCY        22208 non-null  object  
 3   UNIT_ID       22197 non-null  object  
 4   FIRE_NAME     22199 non-null  object  
 5   INC_NUM       21566 non-null  object  
 6   IRWINID       3296 non-null   object  
 7   ALARM_DATE    16865 non-null  object  
 8   CONT_DATE     9632 non-null   object  
 9   C_METHOD      10140 non-null  float64 
 10  CAUSE         22237 non-null  float64 
 11  OBJECTIVE     22014 non-null  float64 
 12  COMPLEX_NAME  608 non-null    object  
 13  COMPLEX_ID    360 non-null    object  
 14  COMMENTS      20034 non-null  object  
 15  FIRE_NUM      17643 non-null  object  
 16  Shape_Length  22261 non-null  float64 
 17  Shape_Area    22261 non-null  float64 
 18

In [4]:
# Check the perimeter CRS 
perimeters.crs

<Projected CRS: EPSG:3310>
Name: NAD83 / California Albers
Axis Info [cartesian]:
- X[east]: Easting (metre)
- Y[north]: Northing (metre)
Area of Use:
- name: United States (USA) - California.
- bounds: (-124.45, 32.53, -114.12, 42.01)
Coordinate Operation:
- name: California Albers
- method: Albers Equal Area
Datum: North American Datum 1983
- Ellipsoid: GRS 1980
- Prime Meridian: Greenwich

In [5]:
# Find the min and max year of the df
print(perimeters.YEAR_.min())
print(perimeters.YEAR_.max())

1878.0
2023.0


#### Data description
This fire perimeter data comes from CalFire and includes data for all fire perimeters from 1878 to 2023. It has data on the year, the fire name, the reporting agency, the cause, duration, among other data. The CRS is NAD83 California Albers and it is a projected CRS (EPSG:3310)

### Select Thomas Fire Boundary 

In [6]:
# Clean all column names
perimeters.columns = perimeters.columns.str.lower()

# Clean year column name
perimeters = perimeters.rename(columns={'year_': 'year'})

In [3]:
# Select 2017 thomas fire boundary
thomas_boundary = perimeters[(perimeters['year'] == 2017.0) & (perimeters['fire_name'] == 'THOMAS')]

# View the data to ensure it looks correct
thomas_boundary.plot()

NameError: name 'perimeters' is not defined

### Export the Thomas boundary as a `GeoJSON`

In [4]:
# Export thomas fire boundary 
thomas_boundary.to_file(filename='data/thomas_boundary.geojson', driver='GeoJSON')

NameError: name 'thomas_boundary' is not defined

I chose to export as a GeoJSON file because it exports as a single file that is straightforward to import in my next jupyter markdown. A GeoJSON is also more compact and has a smaller file size compared to other geospatial data formats. 